python/session-01.md

```diff
@@ -0,0 +1,180 @@
+Implementing these special methods in our classes is a big step to make our code pythonic. Because in this way, we can use built-in functions such as `len()` or index our object with the familiar syntax `obj[key]` and thus our class will need a very brief documentation.    
+
+### Example: A Pythonic Card Deck
+Consider the following code:
+```
+import collections 
+Card = collections.namedtuple('Card', ['rank', 'suit'])
+
+class FrenchDeck:
+    ranks = [str(n) for n in range(2, 11)] + list('JQKA')
+    suits = spades diamonds clubs hearts.split()
+    
+    def __init__(self):
+        self._cards = [Card(rank, suit) for suit in self.suitsfor rank in self.ranks]
+    def __len__(self):
+        return len(self._cards)
+    def __getitem__(self, position):
+        return self._cards[position]
+```
+In the code above, the `__getitem__()` method and `__len__()` method are implemented which enable us to use `len()` and pythonic indexing `obj[key]` while working with the `FrenchDeck` class. This is an example of pythonic coding.  
+
+---
+## Mindomo: a Simple Application For Creating Mind Maps
+![Mindomo](https://www.mindomo.com/blog/wp-content/uploads/2022/06/Mind-mapping-is-....png.webp)
+Mindomo is a very simple application for creating mind maps.
+In order to use Mindomo, you need to create a `.txt`file and write down your mind map. The only syntax for Mindomo is four spaces (a tab). For example:
+```
+The Probelm
+    1. Problem Description
+        Current situation
+            current situation
+        why
+            cause
+        Others
+            outside opinion
+        Ideal situation
+            ideal situation
+    2. Goals
+        goal
+        goal
+    3. Solutions
+        Opportunities
+        Change perspective
+        How might we
+        I wish
+        What if
+        I wonder
+        How to
+    4. Action Plan
+        Actions
+        Actions
+    5. Conclusion
+        Outside opinion
+        Outside opinion
+```
+![Mindomo](https://www.mindomo.com/c/wp-content/uploads/2022/05/3.-Problem-Solving.png)
+You can download Mindomo for free from it's official website
+
+---
+## Obsidian
+Obsidian is simply an interface for plain text files that are stored on your computer, and supports standard Markdown formatting. Many themes will have syntax highlighting to make it easier to see your Markdown-formatted text, but the Markdown that Obsidian supports will be viewable in any other text editor.
+
```

Comment:
Obsidian is a powerful knowledge base that works on top of a local folder of plain text Markdown files. In Obsidian, making and following connections is frictionless, and with the connections in place, you can explore all of your knowledge in the interactive graph view. Obsidian supports CommonMark and GitHub Flavored Markdown (GFM), along with other useful notetaking features such as tags, LaTeX mathematical expressions, mermaid diagrams, footnotes, internal links and embedding Obsidian notes or external files. Obsidian also has a plugin system with 20+ supported plugins to expand its capabilities. The plugin system is currently a work-in-progress, there will be public APIs available to developers in the v1.0 release.
for more information you can visit this <a href="https://obsidian.md/#:~:text=Obsidian%20is%20a%20powerful%20knowledge,%2C%20and%20Linux%20(Flatpak).">link</a>

Reply:
Thanks for the feedback

Reply:
your welcome